### ISSUE TITLE
Question: Is it possible to debug cshtml files in RazorPageGenerator project

### ISSUE THREAD

**Issue by zminic** [aspnet/Razor#1830]
Hi,

I have tried setting breakpoint in cshtml file in RazorPageGenerator project (https://github.com/aspnet/Razor/tree/dev/src/RazorPageGenerator) but breakpoint isn't hit in Visual Studio.

Is there any configuration option to allow this, or this is not possible?

**Comment by NTaylorMullen** [comment]
We don't run the Razor file in the RazorPageGenerator project. However, if we were to I believe that debugging still wouldn't work because we [suppress the checksum](https://github.com/aspnet/Razor/blob/1aff9d0031f8846e836d4050aee6ab9de49b5f12/src/RazorPageGenerator/Program.cs#L127) of the generated Razor view.

**Comment by zminic** [comment]
Hi,

I think that I didn't provide enough context with my question. RazorPageGenerator project generates template c#, then I'm using roslyn compiler to generate assembly on the fly and invoke"ExecuteAsync" method and write result to a file. I'm trying to use it as a replacement for t4 templates but to use razor instead. Final goal is to create dotnet cli extension for template generation.

So far I have managed to create output and I'm satisfied with the results, but I can't get cshtml debugging to work in visual studio.

Thanks for the checksum tip, I will try to remove that suppress feature and see if it works then.

I saw that when I put breakpoint on razor page in aspnetcore mvc visual studio will break on that breakpoint and it will show location of breakpoint to be somethig like _Views_TestView_cshtml. It looks like mvc project creates template class name in that format and maybe visual studio knows to attach debugger if class name matches file name. This is only wild guess, I don't know how debugger internals work on razor, maybe someone can point me to correct repository to search?
